addons/stock/models/stock_move_line.py

```diff
@@ -337,9 +363,10 @@ def create_move(move_line):
         return mls
 
     def write(self, vals):
+        # Very dangerous key. It could create some desynchronization between `stock.move.line` and `stock.quant`.
+        # It exists for performances purposes. Only use it if you handle `stock.quant` reservation manualy.
         if self.env.context.get('bypass_reservation_update'):
-            return super(StockMoveLine, self).write(vals)
-
+            return super().write(vals)
         if 'product_id' in vals and any(vals.get('state', ml.state) != 'draft' and vals['product_id'] != ml.product_id.id for ml in self):
```

Comment:
use `vals.get('product_id')` instead, you never know what can happen with customization : )

Reply:
ah no my bad it already checks `if 'product_id' in vals` nvm for this one